### PR TITLE
Improve wix.targets compatibility with Microsoft.Common.targets

### PR DESCRIPTION
**Pull request by robmen**
Fixes wixtoolset/issues#6628
Fixes wixtoolset/issues#6248
Fixes wixtoolset/issues#5627
Fixes wixtoolset/issues#4607

### CODE REVIEW

File: src/api/api.cmd [hunk 1]
```diff
@@ -11,11 +11,11 @@
...
 :: build
 :: pack
-msbuild -p:Configuration=%_C% api.proj -v:m -m -nr:false || exit /b
+msbuild -p:Configuration=%_C% api.proj -m -nr:false -v:m -nologo || exit /b
 
 :: test
-dotnet test -c %_C% --no-build burn\test\WixToolsetTest.Mba.Core\WixToolsetTest.Mba.Core.csproj || exit /b
-dotnet test -c %_C% --no-build wix\api_wix.sln || exit /b
+dotnet test -c %_C% --no-build --nologo -v m burn\test\WixToolsetTest.Mba.Core\WixToolsetTest.Mba.Core.csproj || exit /b
+dotnet test -c %_C% --no-build --nologo -v m wix\api_wix.sln || exit /b
```
Comment (rseanhall) [inline src/api/api.cmd:18]: I know you already put this in some places earlier, but why do we want minimal verbosity? If something goes wrong, I don't want to have to run another build to find out why.

Comment (barnson) [reply src/api/api.cmd:18]: Error messages still show up. `-v m` just hides the spew from compiler invocations and such.

In a dream world, there's a single MSBuild invocation for the whole world so you could easily specify additional switches (e.g., `/bl`).

Comment (robmen) [reply src/api/api.cmd:18]: I'm still planning to add /bl before I'm done with this PR. I find those are the most enjoyable way to diagnose build failures.

Comment (rseanhall) [reply src/api/api.cmd:18]: I am definitely not looking forward to downloading binary logs to find out why the build failed on the CI server.

Comment (barnson) [reply src/api/api.cmd:18]: They're smaller than diag-level text logs -- I want binlogs for Burn!

**Comment by rseanhall** [comment]
I don't know why @barnson still added the references to `Microsoft.NETFramework.ReferenceAssemblies` in https://github.com/wixtoolset/wix4/pull/54. I had purposely not included them in my VS2022 PR, and removed it from `Directory.Packages.props.pp`, because of these `NU1009` errors. The only purpose I can see for including them is for machines that don't have the .NET Framework targeting pack.

**Comment by barnson** [comment]
Because the GH build failed without them: https://github.com/wixtoolset/wix4/runs/4639436215?check_suite_focus=true

**Comment by robmen** [comment]
Yeah. I can see the appeal of not needing to install targeting packs (especially the old ones) but my patience with Traversal targets ignoring their `DisableImplicitFrameworkReferences` setting is rapidly wearing thin.

**Comment by rseanhall** [comment]
> Because the GH build failed without them: https://github.com/wixtoolset/wix4/runs/4639436215?check_suite_focus=true

Huh? The build failed because you tried to add `Microsoft.NETFramework.ReferenceAssemblies` back to `Directory.Packages.props.pp`: https://github.com/wixtoolset/wix4/commit/5d169aa0710c0557cee8e32067ad1bf3609e6705

**Comment by rseanhall** [comment]
I expected this to fail the build:

> D:\a\wix4\wix4\src\test\burn\WixToolset.WixBA\WixToolset.WixBA.csproj : error NU1009: The packages Microsoft.NETFramework.ReferenceAssemblies are implicitly referenced. You do not typically need to reference them from your project or in your central package versions management file. For more information, see https://aka.ms/sdkimplicitrefs [D:\a\wix4\wix4\src\test\burn\BurnE2ETests.sln]

**Comment by robmen** [comment]
Whoa! So did I. It failed the last "build improvements" PR. I will force revert this one real fast.

**Comment by robmen** [comment]
Oh, thank goodness. It didn't merge. I still need to investigate the root remaining failure in #77.

**Pull request merged**